src/zInit.cpp

```diff
@@ -32,7 +32,22 @@ class LateInitializer {
     }
 
   private:
-    bool checkJvmLoaded() {
+    static bool checkPreload() {
+        Dl_info dlopen_info;
+        Dl_info current_info;
+
+        if (dladdr((const void*)Hooks::init, &current_info) == 0 || current_info.dli_fname == NULL) {
+            return false;
+        }
+
+        if (dladdr((const void*)dlopen, &dlopen_info) == 0 || dlopen_info.dli_fname == NULL) {
```

Comment:
2 things:
- I'd add a comment either here or on the function to explain the flow
- I'd move the declaration of `dlopen_info` on L42